velox/exec/MergeJoin.cpp

```diff
@@ -1237,26 +1294,30 @@ RowVectorPtr MergeJoin::applyFilter(const RowVectorPtr& output) {
 
     // Every time we start a new left key match, `processFilterResult()` will
     // check if at least one row from the previous match passed the filter. If
-    // none did, it calls onMiss to add a record with null right projections to
-    // the output.
+    // none did, it calls onMiss to add a record with null right projections
+    // to the output.
     //
     // Before we leave the current buffer, since we may not have seen the next
-    // left key match yet, the last key match may still be pending to produce a
-    // row (because `processFilterResult()` was not called yet).
+    // left key match yet, the last key match may still be pending to produce
+    // a row (because `processFilterResult()` was not called yet).
     //
     // To handle this, we need to call `noMoreFilterResults()` unless the
-    // same current left key match may continue in the next buffer. So there are
-    // two cases to check:
+    // same current left key match may continue in the next buffer. So there
+    // are two cases to check:
     //
-    // 1. If leftMatch_ is nullopt, there for sure the next buffer will contain
-    // a different key match.
+    // 1. If leftMatch_ is nullopt, there for sure the next buffer will
+    // contain a different key match.
     //
     // 2. leftMatch_ may not be nullopt, but may be related to a different
     // (subsequent) left key. So we check if the last row in the batch has the
     // same left row number as the last key match.
     if (!leftMatch_ || !joinTracker_->isCurrentLeftMatch(numRows - 1)) {
       joinTracker_->noMoreFilterResults(onMiss);
     }
+
+    if (leftMatch_ && !previousLeftMatch_) {
+      joinTracker_->noMoreFilterResults(onMiss);
+    }
```

Comment:
can you comment this? 
we found a bug due to this code @JkSelf

Reply:
@fzhedu It aims to fix the `MergeJoinTest#antiJoinWithFilterWithMultiMatchedRowsInDifferentBatches` test. Although leftMatch is not null, it pertains to the next batch rather than the current one. Therefore, we also need to call `noMoreFilterResults `to update the status of the current batch.

Reply:
@JkSelf here exists a bug: when leftMatch is set a new one, but the output is only contains results from previous leftMatch when compare results 0.

Reply:
@fzhedu Sorry for delay response. Can you help to provide the unit tests that can be reproduced? Thanks.